src/Devices/OWSReadReceiptsProcessor.m

```diff
@@ -90,6 +90,10 @@ - (void)process
             // Mark all unread messages in this thread that are older than message specified in the read
             // receipt.
             NSMutableArray<id<OWSReadTracking>> *interactionsToMarkAsRead = [NSMutableArray new];
+
+            // Always mark the message specified by the read receipt as read.
+            [interactionsToMarkAsRead addObject:message];
```

Comment:
This doesn't seem dangerous or anything, but I'm confused as to how it could fix anything. Can you elaborate on why you made this change?